pyramid/config/security.py

```diff
@@ -218,9 +231,12 @@ def register():
         intr['header'] = header
         intr['safe_methods'] = as_sorted_tuple(safe_methods)
         intr['callback'] = callback
+
+        self.add_subscriber(csrf_token_template_global, [BeforeRender])
```

Comment:
Currently Pyramid has no precedent for automatically registering a `BeforeRender` subscriber and I think we should keep it that way. If you really want to inject it we could a) do it in the current renderer bindings (pyramid_jinja2, pyramid_mako, pyramid_chameleon) or b) do it in the `pyramid.renderer.RendererHelper` as a `system` value.